src/components/custom-aggrid/custom-aggrid-header.js

```diff
@@ -27,15 +50,34 @@ const CustomHeaderComponent = ({
     sortConfig = {},
     onSortChanged,
     updateFilter,
-    filterSelectedOptions = [],
     isSortable = true,
     isFilterable = true,
+    filterParams = {},
+    filterSelector,
 }) => {
-    const { colKey, sortWay } = sortConfig;
-    const isSortActive = colKey === field;
+    const intl = useIntl();
+
+    const {
+        filterUIType = FILTER_UI_TYPES.AUTO_COMPLETE,
+        filterComparators = [],
+        debounceMs,
+    } = filterParams;
 
     const [filterAnchorEl, setFilterAnchorEl] = useState(null);
     const [isHoveringHeader, setIsHoveringHeader] = useState(false);
+    const [selectedFilterComparator, setSelectedFilterComparator] = useState(
+        filterComparators[0]
+    );
+    const [selectedFilterData, setSelectedFilterData] = useState();
```

Comment:
by default it is undefined (nor '', nor [] neither), but I think that it will be cleaner if I change to typescript (considering also your first comment https://github.com/gridsuite/gridstudy-app/pull/1657#discussion_r1380099353)